File: 349.js
// 349. Intersection of Two Arrays
// Given two integer arrays nums1 and nums2, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.

var intersection = function(nums1, nums2) {
    const map = {}
    const arr = []
    for (let num of nums1) {
        map[num] = true
    }
    for (let num of nums2) {
        if (map[num]) {
            arr.push(num)
            map[num] = false
        }
    }
    return arr
};

console.log(intersection([4,9,5], [9,4,9,8,4]))

File: 349.py
# 349. Intersection of Two Arrays
# Given two integer arrays nums1 and nums2, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.

class Solution:
    def intersection(self, nums1: list[int], nums2: list[int]) -> list[int]:
        set1 = set(nums1)
        set2 = set(nums2)
        output = []
        for num in set1:
            if num in set2:
                output.append(num)
        return output
    
s = Solution()
print(s.intersection([1,2,2,1], [2,2]))
print(s.intersection([4,9,5], [9,4,9,8,4]))